cpp/src/transform/transform.cpp

```diff
@@ -0,0 +1,144 @@
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <cudf/utilities/traits.hpp>
+#include <cudf/null_mask.hpp>
+#include <cudf/detail/transform.hpp>
+
+#include <jit/launcher.h>
+#include <jit/type.h>
+#include <jit/parser.h>
+#include "jit/code/code.h"
+
+#include <types.h.jit>
+#include <types.hpp.jit>
+
+namespace cudf {
+namespace experimental {
+namespace transformation {
+
+namespace jit {
+
+
+/**
+ * @brief Functor to enable the internal working of `get_data_ptr`
+ * @ref get_data_ptr
+ */
+struct get_data_ptr_functor {
+  template <typename T>
+  std::enable_if_t<is_fixed_width<T>(), const void *>
+  operator()(column_view const& view) {
+    return static_cast<const void*>(view.template data<T>());
+  }
+  template <typename T>
+  std::enable_if_t<not is_fixed_width<T>(), const void *>
+  operator()(column_view const& view) {
+    CUDF_FAIL("Invalid data type for transform operation");
+  }
+};
+
+/**
+ * @brief Get the raw pointer to data in a (mutable_)column_view
+ */
+auto get_data_ptr(column_view const& view) {
+  return experimental::type_dispatcher(view.type(),
+                         get_data_ptr_functor{}, view);
+}
+
+
+void unary_operation(mutable_column_view output, column_view input,
+                     const std::string& udf, data_type output_type, bool is_ptx,
+                     cudaStream_t stream) {
+ 
+  std::string hash = "prog_transform." 
+    + std::to_string(std::hash<std::string>{}(udf));
+
+  std::string cuda_source;
+  if(is_ptx){
+    cuda_source = "\n#include <cudf/types.hpp>\n" +
+      cudf::jit::parse_single_function_ptx(
+          udf, "GENERIC_UNARY_OP", 
+          cudf::jit::get_type_name(output_type), {0}
+          ) + code::kernel;
+  }else{  
+    cuda_source = 
+      cudf::jit::parse_single_function_cuda(
+          udf, "GENERIC_UNARY_OP") + code::kernel;
+  }
+  
+  // Launch the jitify kernel
+  cudf::jit::launcher(
+    hash, cuda_source,
+    { cudf_types_h, cudf_types_hpp },
+    { "-std=c++14" }, nullptr, stream
+  ).set_kernel_inst(
+    "kernel", // name of the kernel we are launching
+    { cudf::jit::get_type_name(output.type()), // list of template arguments
+      cudf::jit::get_type_name(input.type()) }
+  ).launch(
+    output.size(),
+    get_data_ptr(output),
+    get_data_ptr(input)
+  );
+
+}
+
+} // namespace jit
+} // namespace transformation
+
+
+namespace detail {
+
+std::unique_ptr<column> transform(column_view const& input,
+                                  const std::string &unary_udf,
+                                  data_type output_type, bool is_ptx,
+                                  rmm::mr::device_memory_resource *mr,
+                                  cudaStream_t stream)
+{
+  CUDF_EXPECTS(is_fixed_width(input.type()), "Unexpected non-fixed width type.");
+
+  std::unique_ptr<column> output =
+    make_numeric_column(output_type, input.size(), copy_bitmask(input),
```

Comment:
What about timestamp?

Reply:
Timestamp support would come later when I make a unified type header for jitify. My approach is to include all wrapped types in a single source file and preprocess it to make a single header that can be stringified and included in jitify code. Otherwise we'd have to litter the cmakelists with a stringify of each header we want to import and we'd also have to actively maintain it because those headers cannot include more headers.

Reply:
Okay, in that case update the `is_fixed_width` check above with `is_numeric`.